make/autoconf/jdk-version.m4

```diff
@@ -69,34 +69,38 @@ AC_DEFUN_ONCE([JDKVER_SETUP_JDK_VERSION_NUMBERS],
   AC_SUBST(JDK_RC_PLATFORM_NAME)
   AC_SUBST(HOTSPOT_VM_DISTRO)
 
+  # Note: UTIL_ARG_WITH treats empty strings as valid values when OPTIONAL is false!
+  # That said, it allows for using the RESULT of an option as its own DEFAULT, which
+  # is pretty convenient...
+
+  # Outer [ ] to quote m4.
+  [ USERNAME=`$ECHO "$USER" | $TR -d -c '[a-z][A-Z][0-9]'` ]
+
   # Setup username (for use in adhoc version strings etc)
-  AC_ARG_WITH([build-user], [AS_HELP_STRING([--with-build-user],
-      [build username to use in version strings])])
-  if test "x$with_build_user" = xyes || test "x$with_build_user" = xno; then
-    AC_MSG_ERROR([--with-build-user must have a value])
-  elif test "x$with_build_user" != x; then
-    USERNAME="$with_build_user"
-  else
-    # Outer [ ] to quote m4.
-    [ USERNAME=`$ECHO "$USER" | $TR -d -c '[a-z][A-Z][0-9]'` ]
-  fi
+  UTIL_ARG_WITH(NAME: build-user, TYPE: string,
+    RESULT: USERNAME,
+    DEFAULT: $USERNAME,
+    DESC: [build username to use in version strings],
+    IF_GIVEN: [
```

Comment:
In general, I think these tests would be better expressed using the `CHECK_VALUE` construct, rather than `IF_GIVEN`. (A more specific reason is better than a vaguer and more general)

Reply:
Something concerning is happening when I do that: The code in the CHECK_VALUE block from the resulting configure script has all of the square brackets completely nuked, which is causing the logic in it to always fail. Switching back to IF_GIVEN on my end doesn't help either, it now always destroys all the brackets as well when it was working fine before. Something might be going wrong in the actual utility itself, but I can't tell what exactly :/

Reply:
Is this still a problem after your DEFUN_NAMED fixes?

Reply:
Nope, works spectacularly now ;) I'll switch over to CHECK_VALUE as soon as possible